chains/evm/calls/deploy.go

```diff
@@ -14,48 +14,44 @@ import (
 	"github.com/rs/zerolog/log"
 )
 
-func DeployErc20(c ChainClient, txFabric TxFabric, name, symbol string) (common.Address, error) {
+func DeployErc20(c ClientDeployer, txFabric TxFabric, gasPriceClient GasPricer, name, symbol string) (common.Address, error) {
 	parsed, err := abi.JSON(strings.NewReader(consts.ERC20PresetMinterPauserABI))
 	if err != nil {
 		return common.Address{}, err
 	}
-	address, err := deployContract(c, parsed, common.FromHex(consts.ERC20PresetMinterPauserBin), txFabric, name, symbol)
+	address, err := deployContract(c, parsed, common.FromHex(consts.ERC20PresetMinterPauserBin), txFabric, gasPriceClient, name, symbol)
 	if err != nil {
 		return common.Address{}, err
 	}
 	return address, nil
 }
 
-func DeployBridge(c ChainClient, txFabric TxFabric, domainID uint8, relayerAddrs []common.Address, initialRelayerThreshold *big.Int) (common.Address, error) {
+func DeployBridge(c ClientDeployer, txFabric TxFabric, gasPriceClient GasPricer, domainID uint8, relayerAddrs []common.Address, initialRelayerThreshold *big.Int) (common.Address, error) {
 	parsed, err := abi.JSON(strings.NewReader(consts.BridgeABI))
 	if err != nil {
 		return common.Address{}, err
 	}
-	address, err := deployContract(c, parsed, common.FromHex(consts.BridgeBin), txFabric, domainID, relayerAddrs, initialRelayerThreshold, big.NewInt(0), big.NewInt(100))
+	address, err := deployContract(c, parsed, common.FromHex(consts.BridgeBin), txFabric, gasPriceClient, domainID, relayerAddrs, initialRelayerThreshold, big.NewInt(0), big.NewInt(100))
 	if err != nil {
 		return common.Address{}, err
 	}
 	return address, nil
 }
 
-func DeployErc20Handler(c ChainClient, txFabric TxFabric, bridgeAddress common.Address) (common.Address, error) {
+func DeployErc20Handler(c ClientDeployer, txFabric TxFabric, gasPriceClient GasPricer, bridgeAddress common.Address) (common.Address, error) {
```

Comment:
Same here, if we have a Deployer interface that just has `Deploy` it would be much easier to test with DI and cleaner IMO